### ISSUE TITLE
UnicodeDecodeError: 'utf-8' codec can't decode byte 0x8b in position 6485: invalid start byte

### ISSUE THREAD

**Issue by basicallydean** [dewyman/TranscriptClean#18]
Hi, 

Thanks for the great program, I've had no issues at all running through the pipeline aside from one minor hiccup.

I've corrected 5 datasets with TC - all of which ran perfectly fine, however, one data set (ONT directRNA, mapped with minimap2) is continually outputting this error below. I have tried running this on  a HPC, and two local machines, with different version of TC, but to no avail. 

UnicodeDecodeError: 'utf-8' codec can't decode byte 0x8b in position 6485: invalid start byte 

I have cleaned two other replicates with the same -j -g -v files as I am using for this one. Both were also mapped with the exact same options on minimap2, sorted via samtools. 

I'm having suspicions that there could be something funny going on in my minimap2 output .sam file....

Any help would be appreciated
Cheers
Dean

**Comment by dewyman** [comment]
Hi Dean,
That is certainly odd- I haven't run into that type of error before! Your guess about the Minimap2 SAM file seems plausible. Maybe there is a character in there somewhere that TranscriptClean or one of the libraries it uses is not expecting to be there? Would you mind sending me a sample of the offending SAM file (with header) that causes the problem? Hopefully I can help pinpoint the issue.
All the best,
Dana

**Comment by basicallydean** [comment]
Thanks for the swift response. Yeah that was my reasoning too? I've attached 4000 lines from the .sam file - let me know if the whole thing would help. 
[samplesam.txt](https://github.com/dewyman/TranscriptClean/files/3849552/samplesam.txt)

**Comment by dewyman** [comment]
I ran TranscriptClean on your file in a couple of different modes, and it finished successfully each time. Were you able to reproduce your error with the file you sent me specifically?

**Comment by dewyman** [comment]
Also, I googled your error message, and I'm getting hits related to files being gzipped when programs were not expecting them to be. Can you send me your full TranscriptClean command in case I can spot anything unexpected?
Best,
Dana

**Comment by basicallydean** [comment]
Hi Dana,

I omitted by variant file, thinking that perhaps it was causing the problem given it was the only gzipped input - and voila, the issues disappeared. I noticed that when I ran my other two replicates, I was receiving the "no variant file provided ......" message, despite a .vcf being provided - so it seems the vcf was corrupted. Now for whatever reason the program just didn't want to progress past this . I'm chalking this up to issues with my VCF file and not TranscriptClean so I'll go back and investigate my vcfs as I believe something went wrong when I merged them in the first place. 

Thanks a lot for your quick replies, I'm going to go ahead and close this now.
